src/AutoMerge/guidelines.jl

```diff
@@ -54,11 +66,45 @@ function meets_compat_for_julia(working_directory::AbstractString, pkg, version)
     return false, "There is no compat entry for `julia`."
 end
 
-const guideline_compat_for_all_deps =
-    Guideline("Compat (with upper bound) for all dependencies",
-              data -> meets_compat_for_all_deps(data.registry_head,
-                                                data.pkg,
-                                                data.version))
+const guideline_compat_for_all_deps = Guideline(;
+    info = "Compat (with upper bound) for all dependencies",
+    docs = """
+    Dependencies: All dependencies should have `[compat]` entries that are upper bounded and only include a finite number of breaking releases.
+    For example, the following `[compat]` entries meet the criteria for automatic merging:
+    ```toml
+    [compat]
+    PackageA = "1"          # [1.0.0, 2.0.0), has upper bound (good)
+    PackageB = "0.1, 0.2"   # [0.1.0, 0.3.0), has upper bound (good)
+    ```
+    The following `[compat]` entries do NOT meet the criteria for automatic merging:
+    ```toml
+    [compat]
+    PackageC = ">=3"        # [3.0.0, ∞), no upper bound (bad)
+    PackageD = ">=0.4, <1"  # [-∞, ∞), no lower bound, no upper bound (very bad)
```

Comment:
There are no negative version numbers. Everything is automatically lower bounded at 0.

Reply:
Good catch. Fixed.